packages/main/src/plugin/extension-loader.ts

```diff
@@ -493,21 +493,24 @@ export class ExtensionLoader {
 
   // check if all dependencies are available
   // if not, set the missingDependencies property
-  searchForMissingDependencies(analyzedExtensions: AnalyzedExtension[]): void {
+  async searchForMissingDependencies(analyzedExtensions: AnalyzedExtension[], onStart: boolean): Promise<void> {
+    const existingExtensions = onStart
+      ? analyzedExtensions
+      : [...new Set([...(await this.listExtensions()), ...analyzedExtensions])];
     analyzedExtensions.forEach(extension => {
       const missingDependencies: string[] = [];
       extension.manifest?.extensionDependencies?.forEach((dependency: string) => {
-        if (!analyzedExtensions.find(analyzedExtension => analyzedExtension.id === dependency)) {
+        if (!existingExtensions.find(existingExtension => existingExtension.id === dependency)) {
           missingDependencies.push(dependency);
         }
       });
       extension.missingDependencies = missingDependencies;
     });
   }
 
-  async loadExtensions(analyzedExtensions: AnalyzedExtension[]): Promise<void> {
+  async loadExtensions(analyzedExtensions: AnalyzedExtension[], onStart: boolean): Promise<void> {
```

Comment:
issue(blocking): no parameter needs to be added there

all consumers ask to load extensions, no 'onStart' (or another name paramerer)